src/AuthProvider.tsx

```diff
@@ -34,6 +34,8 @@ let getTokenExpiry = () => {
       return expiresAt;
     } catch (e) {}
   }
+
+  return -1;
 };
```

Comment:
How about this:

* Remove `getTokenExpiry`
* Have `fetchTokenIfNotExpiringSoon` return `{ token, expiresAt }` (assuming the token is not expiring soon)

Then you can just set both `token` and `tokenExpiry` in state at the same time?